packages/elements/src/collapse/index.ts

```diff
@@ -185,24 +198,33 @@ export class Collapse extends BasicElement {
    * @returns clientHeight of the panel so that the panel holder max-height can be set
    */
   private getContentHeight (): number {
-    return this.panel && this.panel.clientHeight || 0;
+    const panelEl = this.panelRef.value;
+    return panelEl ? panelEl.clientHeight : 0;
   }
 
   /**
    * A `TemplateResult` that will be used
    * to render the updated internal template.
-   * @return {TemplateResult}  Render template
+   * @return Render template
    */
   protected render (): TemplateResult {
     return html`
-      <ef-header part="header" level="${this.level}" @tap="${this.handleTap}">
-        <ef-icon icon="right" slot="left" part="toggle"></ef-icon>
-        <slot slot="left" name="header-left"></slot>
-        <slot slot="right" name="header-right"></slot>
-        ${this.header}
+      <ef-header part="header" level="${this.level}">
+        <div part="heading" role="heading" aria-level="${ifDefined(this.headingLevel || undefined)}">
+          <div id="header-toggle"
+               part="header-toggle"
+               role="button"
+               tabindex="0"
+               aria-expanded="${this.expanded}"
+               aria-controls="content"
+               @tap=${this.toggle}>${this.header}</div>
+        </div>
+        <ef-icon icon="right" part="toggle" slot="left" aria-hidden="true"></ef-icon>
+        <slot name="header-left" slot="left"></slot>
```

Comment:
I have discussed with Bangkok Team, the header-left slot should place on the left most of the header because it would be good to custom style for developers. The result should look like this image below, you can try it on the demo page.
![Screen Shot 2022-02-11 at 14 42 31](https://user-images.githubusercontent.com/87511654/153554436-13aeae91-f6b3-4278-988b-a64335708315.png)

Reply:
I have concerns about this as it's an anti-pattern. If you compare the visual design to tree, the slots are in the wrong order - the expand toggle should be on the left.

<img width="722" alt="image" src="https://user-images.githubusercontent.com/1519516/153596676-c6fefabf-8510-4b5f-a20b-2ce040086106.png">

Reply:
Hi Trem, in the use case that you mention the developers can place the checkbox, icon, and text in the main slot instead.